Example/EmbeddedCropViewController.swift

```diff
@@ -59,9 +90,75 @@ class EmbeddedCropViewController: UIViewController {
         }
         return nil
     }
+    
+    override func canPerformAction(_ action: Selector, withSender sender: Any?) -> Bool {
+        
+        if action == #selector(EmbeddedCropViewController.undoButtonPressed(_:)) ||
+           action == #selector(EmbeddedCropViewController.redoButtonPressed(_:))  ||
+           action == #selector(EmbeddedCropViewController.resetButtonPressed(_:)) {
+            
+            guard let toolbarDelegate = toolbarDelegate else { return false }
+
+            return toolbarDelegate.isUndoSupported()
+        }
+        
+        return true
+    }
+    
+    override func validate(_ command: UICommand) {
+        
+        guard let toolbarDelegate = toolbarDelegate else { return }
+        
+        if toolbarDelegate.isUndoSupported() {
+           
+            if command.action == #selector(EmbeddedCropViewController.undoButtonPressed(_:)) {
+                
+                let undoString = NSLocalizedString("Undo", comment: "Undo")
+                
+                command.title = TransformStack.shared.transformDelegate.isUndoEnabled() ? "\(undoString) \(TransformStack.shared.transformDelegate.undoManager().undoActionName)" : undoString
+                
```

Comment:
I think no need to expose TransformStack to users because they don't need to know the implementation details. You can add some proxy functions in `CropViewController` and users only need to talk with `CropViewController`